### ISSUE TITLE
Example in README.md: Lightline.vim could not load colorscheme gruvbox. (While running without `sudo`)

### ISSUE THREAD

**Issue by jirafey** [nix-community/nixvim#1934]
| Field        | Description |
|--------------|-------------|
| Plugin       | `Lightline`  | 
| Nixpkgs      | `24.05` | 
| Home Manager | `24.05` | 

- [x] I have read the [FAQ](https://nix-community.github.io/nixvim/user-guide/faq.html) and my bug is not listed there.

## Minimal, Reproducible Example (MRE)

```nix
  programs.nixvim = {
    enable = true;
    defaultEditor = true;
    colorschemes.gruvbox.enable = true;
    plugins.lightline.enable = true;
  };
}
```

```nix
# home.nix
{ inputs, ... }:

{

  imports = [
    inputs.nixvim.homeManagerModules.nixvim
  ];


  home.username = "user";
  home.homeDirectory = "/home/user";
  # You do not need to change this if you're reading this in the future.
  # Don't ever change this after the first build.  Don't ask questions.
  home.stateVersion = "24.05";

  programs.nixvim = {
    enable = true;
    defaultEditor = true;
    colorschemes.gruvbox.enable = true;
    plugins.lightline.enable = true;
  };
}
```

```nix
# flake.nix
{
  
  inputs = {
    nixpkgs = {
      url = "github:NixOS/nixpkgs?ref=nixos-24.05";
    };
    home-manager = {
      url = "github:nix-community/home-manager?ref=release-24.05";
      inputs.nixpkgs.follows = "nixpkgs";
    };
    nixvim = {
      url = "github:nix-community/nixvim?ref=nixos-24.05";
      inputs.nixpkgs.follows = "nixpkgs";
    };
  };

  outputs = inputs @ { self, nixpkgs, home-manager, ... }: {
    nixosConfigurations = {
      hostname = nixpkgs.lib.nixosSystem {
        system = "x86_64-linux";
        modules = [
          ./configuration.nix

	  home-manager.nixosModules.home-manager {
	    home-manager.extraSpecialArgs = { inherit inputs; };
            home-manager.useGlobalPkgs = true;
	    home-manager.useUserPackages = true;
	    home-manager.users.user = {
	      imports = [
	        # DO NOT IMPORT HOME-MANAGER MODULES HERE
	        ./home.nix
	        ./modules/firefox/firefox.nix
	        ./modules/nvim/nvim.nix
	        ./modules/bash/bash.nix
	      ];
	    };
	  }
        ];
      };
    };
  };
}
```
![image](https://github.com/user-attachments/assets/2ba51dcf-b7bb-4332-8878-1dc3062c8435)

When commenting the line with `lightline` I see gruvbox when using `nvim` without `sudo` but I don't when using it with `sudo`

Also in repositories I have access to, the gruvbox colorscheme is applying with both `sudo` and `non-sudo` `nvim` command. However the error with lightline still appears on `non-sudo` `nvim` command

is there an explanation?

**Comment by jirafey** [comment]
https://github.com/nix-community/nixvim/discussions/1575

**Comment by MattSturgeon** [comment]
> I see gruvbox when using `nvim` without `sudo` but I don't when using it with `sudo`
> 
> Also in repositories I have access to, the gruvbox colorscheme is applying with both `sudo` and `non-sudo` `nvim` command. However the error with lightline still appears on `non-sudo` `nvim` command

This is probably related to how home-manager sets up your environment, I'm not 100% sure ottomh. Usually, `sodo` will inherit your environment though (unless using things like `sudo -i`)

As for lightline, that's strange. We'll have to investigate as it looks like something is broken either upstream in the plugin or in our modules.

**Comment by jirafey** [comment]
> > I see gruvbox when using `nvim` without `sudo` but I don't when using it with `sudo`
> > Also in repositories I have access to, the gruvbox colorscheme is applying with both `sudo` and `non-sudo` `nvim` command. However the error with lightline still appears on `non-sudo` `nvim` command
> 
> This is probably related to how home-manager sets up your environment, I'm not 100% sure ottomh. Usually, `sodo` will inherit your environment though (unless using things like `sudo -i`)
> 
> As for lightline, that's strange. We'll have to investigate as it looks like something is broken either upstream in the plugin or in our modules.

it is rather in my config, i will post the whole thing here soon, i need to fix some issues now.

**Comment by jirafey** [comment]
@MattSturgeon 
https://github.com/jirafey/nixconfig

I would look for some issues with the permissions of the user/sudo

This is a very new config and that is probably why it does not work yet. I probably will abandon nixvim but I am willing to help debug this issue.

**Comment by MattSturgeon** [comment]
I don't see any obvious issues in your config. I suspect this is an issue on our end.

I've not used lightline myself, but my config with lualine seems to be working ok with catppuccin.

**Comment by ech0r** [comment]
manually adding gruvbox under extraPlugins worked for me:

```
    programs.nixvim = {
	enable = true;
        defaultEditor = true;
        extraPlugins = with pkgs.vimPlugins; [
        	gruvbox
        ];
	colorschemes.gruvbox.enable = true;
        plugins.lightline.enable = true;
    };
```

**Comment by MattSturgeon** [comment]
This seems to be the same issue as [this discussion](https://github.com/nix-community/nixvim/discussions/1575), see my response there: https://github.com/nix-community/nixvim/discussions/1575#discussioncomment-10254132

tl;dr, since 871e331722cc7156c04174bfc4ed36f509d5e0fb our gruvbox module installs [`gruvbox-nvim`](https://github.com/ellisonleao/gruvbox.nvim/) which is different to [`gruvbox`](https://github.com/morhetz/gruvbox/). I wonder if lightline only supports the latter and not the one we normally install?

**Comment by jirafey** [comment]
=

> This seems to be the same issue as [this discussion](https://github.com/nix-community/nixvim/discussions/1575), see my response there: [#1575 (reply in thread)](https://github.com/nix-community/nixvim/discussions/1575#discussioncomment-10254132)
> 
> tl;dr, since [871e331](https://github.com/nix-community/nixvim/commit/871e331722cc7156c04174bfc4ed36f509d5e0fb) our gruvbox module installs [`gruvbox-nvim`](https://github.com/ellisonleao/gruvbox.nvim/) which is different to [`gruvbox`](https://github.com/morhetz/gruvbox/). I wonder if lightline only supports the latter and not the one we normally install?

hey if there is any help needed from me, let me know, maybe this issue is ready for close?

**Comment by MattSturgeon** [comment]
@GaetanLepage is our gruvbox maintainer while our lightline module has been [largely untouched](https://github.com/nix-community/nixvim/commits/main/plugins/statuslines/lightline.nix) since it was added 4 years ago.

The `plugins.lightline.colorscheme` option currently defaults to `colorscheme`: https://github.com/nix-community/nixvim/blob/e48ce785d9e72c0106319d93e23c5579336ffe33/plugins/statuslines/lightline.nix#L19-L24

However I'm not 100% sure if the gruvbox module sets the `colorscheme` option :thinking: 


> maybe this issue is ready for close?

Regardless of how well lightline and gruvbox integrate, if the example in the README is broken this is still a valid issue.

**Comment by MattSturgeon** [comment]
> if the example in the README is broken this is still a valid issue.

#2078 didn't address the [README example](https://github.com/nix-community/nixvim?tab=readme-ov-file#what-does-it-look-like), so I'll re-open this.
